### PR TITLE
ISPN-6102 Include the test name in the test thread names

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-6102

Also includes some logging for https://issues.jboss.org/browse/ISPN-6103, but doesn't fix it.

### CODE REVIEW

File: core/src/test/java/org/infinispan/distribution/rehash/ConsistencyStressTest.java [hunk 1]
```diff
@@ -83,14 +83,7 @@ protected void createCacheManagers() throws Throwable {
...
       // create an executor...
-      ExecutorService executorService = Executors.newFixedThreadPool(NUM_NODES * WORKERS_PER_NODE, new ThreadFactory() {
-         int i = 0;
-
-         @Override
-         public synchronized Thread newThread(Runnable r) {
-            return new Thread(r, "Worker-" + i++);
-         }
-      });
+      ExecutorService executorService = Executors.newFixedThreadPool(NUM_NODES * WORKERS_PER_NODE, getTestThreadFactory("Worker"));
```
Comment (pruivo) [inline core/src/test/java/org/infinispan/distribution/rehash/ConsistencyStressTest.java]: you could use fork() for this case IMO. Also, you have the ThreadFactory unused import

**Comment by danberindei** [comment]
Updated, and added one more commit with extra logs for CacheNotifierImplInitialTransferDistTest.

**Comment by pruivo** [comment]
@danberindei ok. running the test and pushing :)

**Comment by pruivo** [comment]
integrated! thanks @danberindei !

**Pull request closed without merging**